libnmstate/dns.py

```diff
@@ -158,7 +158,15 @@ def _find_ifaces_with_static_gateways(self, route_state):
                 if ipv4_iface and ipv6_iface:
                     return (ipv4_iface, ipv6_iface)
                 if route.is_gateway:
-                    if route.is_ipv6:
+                    # Only interfaces not ignored can be considered as valid
+                    # static gateway for nameservers.
+                    iface_ignored = False
+                    for iface in ifaces.all_ifaces():
```

Comment:
Should be `all_kernel_ifaces()` as only kernel interface can hold gateway or having auto-ip.

And `all_kernel_ifaces()` is a hashmap, so you don't need the flag `iface_ignored`.

Reply:
Done, thanks!